expertsystem/data.py

```diff
@@ -126,3 +129,36 @@ def keys(self) -> KeysView[str]:
 
     def values(self) -> ValuesView[Particle]:
         return self.__particles.values()
+
+
+class QuantumState(NamedTuple):
+    """Collection of properties defining a quantum mechanical state.
+
+    Related to `.Particle`, but can contain more specific quantum state
+    information, such as `.Spin` projection.
+    """
+
+    charge: int
+    spin: Spin
+    mass: float
+    strangeness: int = 0
+    charmness: int = 0
+    bottomness: int = 0
+    topness: int = 0
+    baryon_number: int = 0
+    electron_number: int = 0
+    muon_number: int = 0
+    tau_number: int = 0
```

Comment:
Maybe you have some good names for these attributes. electron/muon/tau lepton number are the full physics terms

Reply:
Start to think we need something more like this:
```python
class LeptonNumbers(NamedTuple):
    electron: int = 0
    muon: int = 0
    tau: int = 0

class QuarkNumbers(NamedTuple):
    strange: int = 0
    charm: int = 0
    bottom: int = 0
    top: int = 0
```

Reply:
Hmm, but what is the gain of using that? Btw isospin is also a quark flavor attribute

Reply:
Just to group things. In practical terms, this would only make it more complicated to construct a `Particle` or `QuantumState` class...